tests/test_proxy.py

```diff
@@ -0,0 +1,53 @@
+import os
+import shutil
+import sys
+import tempfile
+
+import pytest
+
+from pressurecooker import proxy
+from pressurecooker.youtube import YouTubeResource
+
+
+# This test takes a few minutes, but is very useful for checking that the proxy is not being ignored,
+# so mark it to run when the PYTEST_RUN_SLOW env var is set.
+@pytest.mark.skipif(not 'PYTEST_RUN_SLOW' in os.environ, reason="This test takes several minutes to complete.")
+def test_bad_proxies_get_banned():
+    # create some fake proxies, then check that they get banned.
+    proxy.PROXY_LIST = [
+        '123.123.123.123:1234',
+        '142.123.432.234:123345',
+        '156.245.233.321:1232342',
+        '444.444.444.444:445565',
+        '555.555.555.555:5454'
+    ]
+
+    temp_dir = tempfile.mkdtemp()
+    video = YouTubeResource('https://www.youtube.com/watch?v=DLzxrzFCyOs')
+    video.download(temp_dir)
+
+    temp_files = os.listdir(temp_dir)
+    shutil.rmtree(temp_dir)
+    # if the file downloaded, that's a sign the proxies weren't used.
+    assert temp_files == ['Watch']
```

Comment:
This check doesn't work as expected, since files get saved inside the `Watch` dir:

```
/var/folders/wc/2r44j8gs4gn56t1xtw5f6wlm0000gn/T/tmpQkBDUs
└── Watch
    ├── DLzxrzFCyOs.jpg
    └── DLzxrzFCyOs.mp4
```

So maybe check should be to listdir inside `Watch/` and check it's an empty list?